parsers/CL.py

```diff
@@ -59,6 +70,38 @@ def data_processor(raw_data):
     return ordered_data
 
 
+def get_data_live(session, logger):
+    """Requests generation data in json format."""
+
+    s = session or requests.session()
+    json_data = s.get(API_BASE_URL_LIVE).json()
+    return json_data
+
+
+def production_processor_live(json_data, zone_key):
+    """
+    Extracts data timestamp and sums regional data into totals by key.
+    Maps keys to type and returns a tuple.
+    """
+
+# unsure about this time object, but seems to be unix time, but in msecs
+        # 1) is this right?
+        # 2) does this need to have a timezone or anything??
+    dt = arrow.get(json_data['fecha']/1000)
+    
+    totals = defaultdict(lambda: 0.0)
+    
+    breakdown = json_data['data']
+
+    for generation in breakdown:
+        totals[generation['key']] = generation['value']
+
+    mapped_totals = {TYPE_MAPPING_LIVE.get(name, 'unknown'): val for name, val
+                     in totals.items()}
```

Comment:
We should probably raise a warning in the logger if a new energy type appears.

Reply:
In looking at doing this, I've got myself confused. If there is an unexpected source, then either the "TERMICAS" data or that of the unexpected one will be overwritten.

In fact I realised this was already happening, the TERMICAS data was being overwritten by the total production.

I've got round this by adding the total to TYPE_MAPPING_LIVE and using this to check for 'unexpected' values, which now don't overwrite 'unknown'. I then remove the total.

What do you think?

Reply:
This may be irrelevant now...